test/pool/TestPoolIntegration.spec.ts

```diff
@@ -782,15 +917,108 @@ contract("Pool", async (accounts) => {
+        const NEW_RATE_BAT = ONE_USD_IN_SCALE.mul(new BN(110)).div(new BN(100));
+        await priceOracle.setPrice(cBAT_addr, NEW_RATE_BAT);
+
+        const debt = await bBAT.borrowBalanceCurrent.call(a.user1);
+        const debtTopupInfo = await pool.getDebtTopupInfo.call(a.user1, bBAT_addr);
+        const expectedMinTopup = debt.mul(minTopupBps).div(new BN(10000));
+        const expectedMaxTopup = debt.div(await pool.membersLength());
+        expectDebtTopupInfo(debtTopupInfo, {
+          expectedMinTopup: expectedMinTopup,
+          expectedMaxTopup: expectedMaxTopup,
+          expectedIsSmall: true,
+        });
+
+        // member deposit
+        await BAT.approve(pool.address, expectedMinTopup, { from: a.member1 });
+        await pool.methods["deposit(address,uint256)"](BAT_addr, expectedMinTopup, {
+          from: a.member1,
+        });
+
+        // topup
+        await pool.topup(a.user1, bBAT_addr, expectedMinTopup, false, {
+          from: a.member1,
+        });
+
+        expect(await avatar1.canLiquidate.call()).to.be.equal(true);
+        expect(await avatar1.canUntop.call()).to.be.equal(false);
+
+        // partial liquidation
+        const debtToLiquidate = await avatar1.getMaxLiquidationAmount.call(cBAT_addr);
+        const result = await avatar1.calcAmountToLiquidate.call(cBAT_addr, debtToLiquidate);
+        const amtToDeductFromTopup = result["amtToDeductFromTopup"];
+        const amtToRepayOnCompound = result["amtToRepayOnCompound"];
+        const halfDebtToLiquidate = debtToLiquidate.div(new BN(2));
+
+        // member deposit
+        await BAT.approve(pool.address, amtToRepayOnCompound, { from: a.member1 });
+        await pool.methods["deposit(address,uint256)"](BAT_addr, amtToRepayOnCompound, {
+          from: a.member1,
+        });
+
+        await pool.liquidateBorrow(a.user1, bETH_addr, bBAT_addr, halfDebtToLiquidate, {
+          from: a.member1,
+        });
+
+        // validate deposit & topup balance
+        expect(await pool.balance(a.member1, BAT_addr)).to.be.bignumber.equal(
+          amtToRepayOnCompound.div(new BN(2)),
+        );
+        expect(await BAT.balanceOf(pool.address)).to.be.bignumber.equal(
+          amtToRepayOnCompound.div(new BN(2)),
+        );
+        expect(await pool.topupBalance(a.member1, BAT_addr)).to.be.bignumber.equal(
+          expectedMinTopup.div(new BN(2)),
+        );
+
+        expect(await avatar1.isPartiallyLiquidated()).to.be.equal(true);
+
+        // user repay
+        await BAT.approve(bBAT_addr, ONE_HUNDRED_BAT, { from: a.user1 });
+        await bBAT.repayBorrow(ONE_HUNDRED_BAT, { from: a.user1 });
+
+        expect(await avatar1.canLiquidate.call()).to.be.equal(false);
+        expect(await avatar1.canUntop.call()).to.be.equal(true);
+
+        // member attempt liquidateBorrow, must fail
+        await expectRevert(
+          pool.liquidateBorrow(a.user1, bETH_addr, bBAT_addr, halfDebtToLiquidate, {
```

Comment:
should try with user paying only small part of his debt, and then liquidator tries to pay only small portion from the debt.
with current test it might fail because there is not enough debt to liquidate.